crates/swift-bridge-ir/src/bridged_type/shared_struct.rs

```diff
@@ -19,6 +19,7 @@ pub(crate) struct SharedStruct {
     pub fields: StructFields,
     pub swift_name: Option<LitStr>,
     pub already_declared: bool,
+    pub derives: Option<Vec<TokenStream>>,
```

Comment:
Instead of `Option<Vec<TokenStream>>` let's use something like:

```rust
pub(crate) struct StructDerives {
    pub copy: bool,
    pub clone: bool,
}
```

Reply:
This would kind of limit us to only these kind of derives, if we want to add new ones (like `Debug`) we would need to open a PR and add them. It also has the potential of growing quite a lot. I don't see why it would be an issue to have the more generic `Vec<TokenStream>`?

Reply:
Thanks for explaining your thinking. Makes sense.

---

The issue is that we only want to support derives that we know are well behaved. 
Supporting arbitrary derives might be possible, but requires more research.

As it stands now, supporting arbitrary derives can lead to undefined behavior. 
It's incredibly unlikely, but theoretically possible. 

Before we support arbitrary derives we want to make UB *completely* impossible.

---

Essentially, `#[derive(SomeCustomDerive)]` could emit something like `type u32 = some_crate::NotAU32` and cause `swift-bridge` to think that it sees a `u32` not realizing that a type alias got generated to make `u32` a completely different type.

It should be possible to guard against this, but until we guard against it (and any other potential derive related UB scenarios) we can't allow arbitrary derives.

This discussion has more detail https://github.com/chinedufn/swift-bridge/issues/190#issuecomment-1462315921

Reply:
makes sense, will go with your suggestion then

Reply:
[ead0af9](https://github.com/chinedufn/swift-bridge/pull/198/commits/ead0af99013d5cafb5e2cb8d0da6ba7a191d8211)